hs/src/Reach/Simulator/Core.hs

```diff
@@ -12,13 +12,33 @@ import Data.Bits
 import Data.Aeson (FromJSON, ToJSON)
 import GHC.Generics
 
--- state
+data LocalState = LocalState
+  { l_acct :: Account
+  , l_who :: Maybe SLPart
+  -- TODO QUESTION: partstates for each participant?
+  , l_init_val :: DLVal
+  }
+
+-- initLocalState :: LocalState
+-- initLocalState = LocalState
+--   { l_acct = -1
+--   , l_who = Nothing
+--   }
+
+type PartId = Int
+type Locals = M.Map PartId LocalState
+
+-- TODO: state here and in the server should
+-- probably all be IORefs
```

Comment:
I don't think so because then you'll have to find and duplicate them all. It's a trade-off... do you had complexity trying to duplicate, or do you add complexity functionally updating them? My gut says the first is riskier